client/vaultclient/vaultclient.go

```diff
@@ -157,6 +159,10 @@ func NewVaultClient(config *config.VaultConfig, logger hclog.Logger, tokenDerive
 		return nil, err
 	}
 
+	client.SetHeaders(http.Header{
+		"User-Agent": []string{"HashiCorp/nomad"},
```

Comment:
lower case `hashicorp`?

Reply:
Will do - noticed that some other hashicorp libraries use this casing (e.g. [1]), so went with it for consistency, but lower casing makes more sense to me

[1] https://github.com/hashicorp/go-checkpoint/blob/1545e56e46dec3bba264e41fde2c1e2aa65b5dd4/checkpoint.go#L128